cli/src/cmd/inspect.rs

```diff
@@ -0,0 +1,249 @@
+    pub contract: String,
+
+    #[clap(help = "the contract artifact field to inspect")]
+    pub mode: ContractArtifactFields,
+
+    /// All build arguments are supported
+    #[clap(flatten)]
+    build: build::BuildArgs,
+}
+
+impl Cmd for InspectArgs {
+    type Output = ();
+    fn run(self) -> eyre::Result<Self::Output> {
+        let InspectArgs { contract, mode, build } = self;
+
+        // Map mode to ContractOutputSelection
+        let mut cos = build.compiler.extra_output.unwrap_or_default();
+        if !cos.iter().any(|&i| i.to_string() == mode.to_string()) {
+            match mode {
+                ContractArtifactFields::Abi => cos.push(ContractOutputSelection::Abi),
+                ContractArtifactFields::Bytecode => { /* Auto Generated */ }
+                ContractArtifactFields::DeployedBytecode => { /* Auto Generated */ }
+                ContractArtifactFields::Assembly | ContractArtifactFields::AssemblyOptimized => {
+                    cos.push(ContractOutputSelection::Evm(EvmOutputSelection::Assembly))
+                }
+                ContractArtifactFields::MethodIdentifiers => {
+                    cos.push(ContractOutputSelection::Evm(EvmOutputSelection::MethodIdentifiers))
+                }
+                ContractArtifactFields::GasEstimates => {
+                    cos.push(ContractOutputSelection::Evm(EvmOutputSelection::GasEstimates))
+                }
+                ContractArtifactFields::StorageLayout => {
+                    cos.push(ContractOutputSelection::StorageLayout)
+                }
+                ContractArtifactFields::DevDoc => cos.push(ContractOutputSelection::DevDoc),
+                ContractArtifactFields::Ir => cos.push(ContractOutputSelection::Ir),
+                ContractArtifactFields::IrOptimized => {
+                    cos.push(ContractOutputSelection::IrOptimized)
+                }
+                ContractArtifactFields::Metadata => cos.push(ContractOutputSelection::Metadata),
+                ContractArtifactFields::UserDoc => cos.push(ContractOutputSelection::UserDoc),
+                ContractArtifactFields::Ewasm => {
+                    cos.push(ContractOutputSelection::Ewasm(EwasmOutputSelection::All))
+                }
+            }
+        }
+
+        // Run Optimized?
+        let optimized = if let ContractArtifactFields::AssemblyOptimized = mode {
+            true
+        } else {
+            build.compiler.optimize
+        };
+
+        // Build modified Args
+        let modified_build_args = BuildArgs {
+            compiler: CompilerArgs {
+                extra_output: Some(cos),
+                optimize: optimized,
+                ..build.compiler
+            },
+            ..build
+        };
+
+        // Build the project
+        let project = modified_build_args.project()?;
+        let outcome = super::suppress_compile(&project)?;
+
+        // Find the artifact
+        let found_artifact = outcome.find(contract.clone());
```

Comment:
this seems unecessary, this should also work
```suggestion
        let found_artifact = outcome.find(&contract);
```